### DISCUSSION TITLE
split PRs per package.json

### DISCUSSION THREAD

**Discussion by satanTime** [renovatebot/renovate#7993] [Request Help] [answered]
Hi,

thanks for the app, it's a really cool one.

However, I'm stuck with an issue.
I have a monorepo that should contain Angular 5, 6, 7, etc. for e2e tests.
Therefore I want to avoid group pull requests which update several Angulars at once.
I want that 1 PR contains changes for 1 package.json and its lock file only.

I tried to disable renovate globally and to use `packageRules` enabling it per group, but looks like it still groups all the groups.
The config is here: https://github.com/satanTime/ng-mocks/blob/master/renovate.json

Thank you in advance!

**Comment by viceice** [comment]
https://docs.renovatebot.com/templates/

Try baseDir and group in package rules.

**Comment by satanTime** [comment]
Thanks for the hint, unfortunately, no luck.

I've tried
```
    {
      "enabled": true,
      "groupName": "a11",
      "baseDir": "e2e/a11",
      "paths": ["+(package.json)", "+(package-lock.json)"]
    }
```
and

```
    {
      "enabled": true,
      "groupName": "a11",
      "baseDir": "+(e2e/a11)",
      "paths": ["+(e2e/a11/package.json)", "+(e2e/a11/package-lock.json)"]
    }
```

but still PRs are grouped.

**Reply by viceice** [reply]
You need different group names, as it will be used to create the branch name. You can use the baseDir in group template.

**Reply by satanTime** [reply]
I have them, I meant, I had tried different values for `baseDir` and `paths`.

```typescript

    {
      "enabled": true,
      "groupName": "a10",
      "baseDir": "e2e/a10",
      "paths": ["+(package.json)", "+(package-lock.json)"]
    },
    {
      "enabled": true,
      "groupName": "a11",
      "baseDir": "e2e/a11",
      "paths": ["+(package.json)", "+(package-lock.json)"]
    }
```

Unfortunately, in docs it says that "group" is an object and used internally by renovate, but how to configure it isn't clear.

**Reply by satanTime** [reply]
It was a good hint, I think I need

`"additionalBranchPrefix": "{{parentDir}}-"` and no groups.

**Reply by viceice** [reply]
You can apply this with a package rule to exclude the root package.json

**Comment by satanTime** [comment, marked as answer]
The answer on my question is usage of `additionalBranchPrefix`.

`"additionalBranchPrefix": "{{#if baseDir}}{{baseDir}}/{{else}}root/{{/if}}"`

`baseDir` in my case is:
`<empty>` - root package.json
`e2e/a5` - angular 5
`e2e/a6` - angular 6

**Reply by dudicoco** [reply]
thanks @satanTime, very helpful for my use case
